src/main/java/net/blay09/mods/waystones/block/WaystoneBlock.java

```diff
@@ -46,29 +53,45 @@
 import java.util.Objects;
 import java.util.Random;
 
-public class WaystoneBlock extends Block {
+public class WaystoneBlock extends Block implements IWaterLoggable {
 
-    /**
-     * We provide a slightly smaller render shape to prevent neighbour blocks from being culled.
-     */
-    private static final VoxelShape RENDER_SHAPE = VoxelShapes.create(1 / 16f, 1 / 16f, 1 / 16f, 15 / 16f, 15 / 16f, 15 / 16f);
+    // Precise selection box
+    private static final VoxelShape WAYSTONE_UPPER = VoxelShapes.or(
+      Block.makeCuboidShape(3, 0, 3, 13, 8, 13),
+      Block.makeCuboidShape(2, 8, 2, 14, 10, 14),
+      Block.makeCuboidShape(1, 10, 1, 15, 12, 15),
+      Block.makeCuboidShape(3, 12, 3, 13, 14, 13),
+      Block.makeCuboidShape(4, 14, 4, 12, 16, 12))
+      .simplify();
+
+    private static final VoxelShape WAYSTONE_LOWER = VoxelShapes.or(
+      Block.makeCuboidShape(0, 0, 0, 16, 3, 16),
+      Block.makeCuboidShape(1, 3, 1, 15, 7, 15),
+      Block.makeCuboidShape(2, 7, 2, 14, 9, 14),
+      Block.makeCuboidShape(3, 9, 3, 13, 16, 13))
+      .simplify();
 
     public static final DirectionProperty FACING = BlockStateProperties.HORIZONTAL_FACING;
     public static final EnumProperty<DoubleBlockHalf> HALF = BlockStateProperties.DOUBLE_BLOCK_HALF;
+    private static final BooleanProperty WATERLOGGED = BlockStateProperties.WATERLOGGED;
 
     public WaystoneBlock() {
         super(Properties.create(Material.ROCK).sound(SoundType.STONE).hardnessAndResistance(5f, 2000f));
+        this.setDefaultState(this.stateContainer.getBaseState().with(WaystoneBlock.WATERLOGGED, false));
     }
 
     @Override
-    public VoxelShape getRenderShape(BlockState p_196247_1_, IBlockReader p_196247_2_, BlockPos p_196247_3_) {
-        return RENDER_SHAPE;
+    public VoxelShape getShape(BlockState p_196247_1_, IBlockReader p_196247_2_, BlockPos p_196247_3_, final ISelectionContext context) {
+        final DoubleBlockHalf half = p_196247_1_.get(WaystoneBlock.HALF);
```

Comment:
`p_196247_1_` should be given a sensible name  (e.g. state)

Reply:
Alright, thats fixed